src/System.Text.Json/src/System/Text/Json/Serialization/JsonSnakeCaseNamingPolicy.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSnakeCaseNamingPolicy : JsonNamingPolicy
+    {
+        public override string ConvertName(string name)
+        {
+            if (string.IsNullOrEmpty(name))
+                return name;
+
+            // Allocates a string builder with the guessed result length,
+            // where 5 is the average word length in English, and
+            // max(2, length / 5) is the number of underscores.
+            StringBuilder builder = new StringBuilder(name.Length + Math.Max(2, name.Length / 5));
+            UnicodeCategory? previousCategory = null;
+
+            for (int currentIndex = 0; currentIndex < name.Length; currentIndex++)
+            {
+                char currentChar = name[currentIndex];
+                if (currentChar == '_')
+                {
+                    builder.Append('_');
+                    previousCategory = null;
+                    continue;
+                }
+
+                UnicodeCategory currentCategory = char.GetUnicodeCategory(currentChar);
```

Comment:
@ahsonkhan Out of curiosity, when method already contains the return type name is it okay to use `var` on the left side?

Reply:
I don't think so (but it is a bit of a grey zone). Use `var` when the RHS has `new` or otherwise it is clear what the type is from reading that line (without having prior knowledge of what the calling API is doing).

> We only use `var` when it's obvious what the variable type is

I can see the argument that GetX methods return X and hence it is clear, so this comes down to what is considered "obvious" (and probably personal preference). What if GetX was returning an interface/abstract type (like IX)?